src/clustering.jl

```diff
@@ -7,13 +7,26 @@ function get_colorscheme(
     tol=Clustering._kmeans_default_tol,
 )::Vector{Lab}
     # Cluster in Lab color space
-    data = reshape(channelview(Lab.(img)), 3, :)
+
+    # Clustering on the downsampled image already generates good enough colormap estimation
+    # This significantly reduces the algorithmic complexity.
+    img = _restrict_to(img, ncolors*100)
+    data = convert(Array{Float32}, reshape(channelview(Lab.(img)), 3, :))
```

Comment:
For many big-data problems, it's often a good strategy to eagerly collect data into a contiguous memory layout so that future slicing on `data[inds...]` can be faster.
I added this before trying the downsample trick only to verify if this helps reduce the allocations. Here I don't observe any performance difference, I guess it's because Clustering.jl is doing a similar thing on their side.

I've reverted this in the new commit.